### ISSUE TITLE
GenerateTangents in 0.14.1 can panic even though it is supposed to return Result 

### ISSUE THREAD

**Issue by ethereumdegen** [bevyengine/bevy#14874]
## Bevy version
 0.14.1 
 
## What went wrong

MESH:: GenerateTangents() in 0.14.1 can panic even though it is supposed to return Result 

```
 thread 'Compute Task Pool (1)' panicked at /home/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_mikktspace-0.14.0/src/generated.rs:1033:15:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_clay_tiles::clay_tile_block::build_tile_block_meshes`!
```

**Comment by rparrett** [comment]
Are you able to provide the mesh that is causing this panic?

**Comment by ethereumdegen** [comment]
not easily no because it could be one of many and it was one generated by a procgen mesh algo  :( 

this algo https://github.com/ethereumdegen/bevy_clay_tiles/blob/master/src/pre_mesh.rs

**Comment by rparrett** [comment]
Even a `println!("{:?}", mesh` would probably be useful.

**Comment by ethereumdegen** [comment]
```

**Comment by ethereumdegen** [comment]
```
{ primitive_topology: TriangleList, attributes: {MeshVertexAttributeId(0): MeshAttributeData { attribute: MeshVertexAttribute { name: "Vertex_Position", id: MeshVertexAttributeId(0), format: Float32x3 }, values: Float32x3([[0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]) }, MeshVertexAttributeId(1): MeshAttributeData { attribute: MeshVertexAttribute { name: "Vertex_Normal", id: MeshVertexAttributeId(1), format: Float32x3 }, values: Float32x3([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]) }, MeshVertexAttributeId(2): MeshAttributeData { attribute: MeshVertexAttribute { name: "Vertex_Uv", id: MeshVertexAttributeId(2), format: Float32x2 }, values: Float32x2([[0.0, 1.0], [0.0, 0.0], [0.0, 1.0], [0.0, 1.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0], [0.0, 0.0], [0.0, 1.0], [0.0, 1.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0], [0.0, 0.0], [0.0, 1.0], [0.0, 1.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0], [0.0, 0.0], [0.0, 1.0], [0.0, 1.0], [0.0, 0.0], [0.0, 0.0], [0.0, 1.0], [0.0, 0.0], [0.0, 1.0], [0.0, 1.0], [0.0, 0.0], [0.0, 0.0]]) }}, indices: Some(U32([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29])), morph_targets: None, morph_target_names: None, asset_usage: RenderAssetUsages(MAIN_WORLD | RENDER_WORLD) }
thread 'Compute Task Pool (1)' panicked at /home/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_mikktspace-0.14.0/src/generated.rs:1033:15:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_clay_tiles::clay_tile_block::build_tile_block_meshes`!

````

**Comment by kumikaya** [comment]
The panic occurs in the function `genTangSpace` due to a scenario where the number of valid triangles (`iNrTrianglesIn`) becomes zero after degenerate triangle detection and removal. This leads to the `InitTriInfo` function being called with zero triangles, causing a panic.

**Comment by ethereumdegen** [comment]
Ohhh lol its probably a mesh with a width of 0.0 .  So basically an invalid mesh like a plane. But still

**Comment by rparrett** [comment]
As I understand it, `bevy_mikktspace` may not be in a place where we can really fix stuff like this until #9050 gets reviewed/merged.
